components/nimbus/src/lib.rs

```diff
@@ -43,3 +53,85 @@ pub use evaluator::TargetingAttributes;
 
 #[cfg(test)]
 mod tests;
+
+impl UniffiCustomTypeConverter for JsonObject {
+    type Builtin = String;
+
+    fn into_custom(val: Self::Builtin) -> uniffi::Result<Self> {
+        let json: Value = serde_json::from_str(&val)?;
+
+        match json.as_object() {
+            Some(obj) => Ok(obj.clone()),
+            _ => Err(anyhow::anyhow!(
+                "Unexpected JSON-non-object in the bagging area"
+            )),
+        }
+    }
+
+    fn from_custom(obj: Self) -> Self::Builtin {
+        Value::Object(obj).to_string()
+    }
+}
+
+cfg_if::cfg_if! {
+    if #[cfg(feature = "stateful-uniffi-bindings")] {
+        include!(concat!(env!("OUT_DIR"), "/nimbus.uniffi.rs"));
```

Comment:
I think let's move these `include!` calls to their respective `nimbus_client.rs` and `cirrus_client.rs` files, and move the custom  converters to their own files.

Reply:
It seems if I move these into their respective `client` files, then I have to write the definition for the `JsonObject` type converter in each of those files as well. The `UniffiCustomTypeConverter` is private so it's only available to the file the scaffolding rust is included in.